client/lib/plans/index.js

```diff
@@ -66,6 +68,28 @@ export function getSitePlanSlug( siteID ) {
 	return get( site, 'plan.product_slug' );
 }
 
+function canUpgradeToPlan( planKey, site ) {
+	const offer = site.jetpack ? JETPACK : WPCOM;
+	return get( plansList, [ planKey, 'availableIn' ], () => false )( offer );
+}
+
+export function getUpgradePlanSlugFromPath( path, siteID ) {
+	const site = siteID ? sitesList.getSite( siteID ) : sitesList.getSelectedSite();
+	return find( Object.keys( plansList ), planKey => (
+		( planKey === path || getPlanPath( planKey ) === path ) &&
+		canUpgradeToPlan( planKey, site )
+	) );
+}
+
+export function getPlanPath( plan ) {
+	return get( plansList, [ plan, 'getPathSlug' ], () => undefined )();
+}
+
+export const getCartItem = ( productSlug, isFreeTrial = false ) => ( {
```

Comment:
We can keep this function, but I would still put this logic closer to the checkout action since its unlikely to be reused and it doesn't rely on any of the constants in this file.